src/bzflag/HUDRenderer.cxx

```diff
@@ -1716,28 +1649,20 @@ void HUDRenderer::drawMarkersInView( int centerx, int centery, const LocalPlayer
     {
         glPushMatrix();
 
-        hudColor3Afv( hudColor, 0.5f );
-
         glTranslatef((float)centerx,(float)centery,0);
         glLineWidth(2.0f);
 
+        auto viewPos = glm::make_vec2(myTank->getPosition());
+
         // draw any waypoint markers
-        for (int i = 0; i < (int)enhancedMarkers.size(); i++)
-        {
-            drawWaypointMarker(glm::value_ptr(enhancedMarkers[i].color), 0.45f,
-                               glm::value_ptr(enhancedMarkers[i].pos), myTank->getPosition(),
-                               enhancedMarkers[i].name, enhancedMarkers[i].friendly);
-        }
+        for (auto enhancedMarker : enhancedMarkers)
```

Comment:
a note on range-based `for` (and `auto`)

Whereas a "normal" `for` loop would operate on iterators, the range-based version operates on elements of the container. So for
```c++
std::vector<Foo> container;
for (auto item : container) {...}
```
`item` is a _copy_ of each element in the container. If the element is trivially copyable (and you don't want to modify it), this may be acceptable.

If you do want to modify it, be sure to use `auto& item : container`.

If it's not trivially copyable and you don't want to modify it, use `auto const& item : container`

Also, remember that if you are iterating over a `std::map`, each element is actually a `std::pair<key_type,mapped_type>` so you might need to access `item.first` (key) or `item.second` (value)